flink-python/src/test/java/org/apache/flink/client/python/PythonFunctionSqlTest.java

```diff
@@ -0,0 +1,120 @@
+import org.apache.flink.util.FileUtils;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.UUID;
+
+import static org.apache.flink.python.PythonOptions.PYTHON_FILES;
+
+/**
+ * Tests for PythonFunctionSqlTest.
+ */
+public class PythonFunctionSqlTest {
+
+	private String tmpdir = "";
+	private BatchTableEnvironment flinkTableEnv;
+	private StreamTableEnvironment blinkTableEnv;
+	private Table flinkSourceTable;
+	private Table blinkSourceTable;
+
+	private void closeStartedPythonProcess() {
+		synchronized (PythonFunctionFactoryUtil.class) {
+			if (PythonFunctionFactoryUtil.pythonProcessShutdownHook != null) {
+				PythonFunctionFactoryUtil.pythonProcessShutdownHook.run();
+				Runtime.getRuntime().removeShutdownHook(PythonFunctionFactoryUtil.pythonProcessShutdownHook);
+				PythonFunctionFactoryUtil.pythonProcessShutdownHook = null;
+			}
+		}
+	}
+
+	@Before
+	public void prepareEnvironment() throws IOException {
+		closeStartedPythonProcess();
+		tmpdir = new File(System.getProperty("java.io.tmpdir"), UUID.randomUUID().toString()).getAbsolutePath();
+		new File(tmpdir).mkdir();
+		File pyFilePath = new File(tmpdir, "test1.py");
+		try (OutputStream out = new FileOutputStream(pyFilePath)) {
+			String code = ""
+				+ "from pyflink.table.udf import udf\n"
+				+ "from pyflink.table import DataTypes\n"
+				+ "@udf(input_types=DataTypes.STRING(), result_type=DataTypes.STRING())\n"
+				+ "def func1(str):\n"
+				+ "    return str + str\n";
+			out.write(code.getBytes());
+		}
+		ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+		flinkTableEnv = BatchTableEnvironment.create(env);
+		flinkTableEnv.getConfig().getConfiguration().set(PYTHON_FILES, pyFilePath.getAbsolutePath());
+		StreamExecutionEnvironment sEnv = StreamExecutionEnvironment.getExecutionEnvironment();
+		blinkTableEnv = StreamTableEnvironment.create(
+			sEnv, EnvironmentSettings.newInstance().useBlinkPlanner().inStreamingMode().build());
+		blinkTableEnv.getConfig().getConfiguration().set(PYTHON_FILES, pyFilePath.getAbsolutePath());
+		flinkSourceTable = flinkTableEnv.fromDataSet(env.fromElements("1", "2", "3")).as("str");
+		blinkSourceTable = blinkTableEnv.fromDataStream(sEnv.fromElements("1", "2", "3")).as("str");
+	}
+
+	@After
+	public void cleanEnvironment() throws IOException {
+		closeStartedPythonProcess();
+		FileUtils.deleteDirectory(new File(tmpdir));
+	}
+
+	@Test
+	public void testPythonFunctionFactory() {
+		// flink temporary catalog
+		flinkTableEnv.sqlUpdate("create temporary function func1 as 'test1.func1' language python");
+		Table flinkTable = flinkSourceTable.select("func1(str)");
+		String flinkPlan = flinkTableEnv.explain(flinkTable);
+		Assert.assertTrue(flinkPlan.contains("DataSetPythonCalc(select=[func1(f0) AS _c0])"));
```

Comment:
Change `DataSetPythonCalc` to `PythonCalc` and refactor the test case a bit to remove the code duplication?